packager/mpd/base/period.cc

```diff
@@ -260,7 +260,38 @@ void Period::ProtectedAdaptationSetMap::Register(
 
 bool Period::ProtectedAdaptationSetMap::Match(
     const AdaptationSet& adaptation_set,
-    const MediaInfo& media_info) {
+    const MediaInfo& media_info,
+    bool content_protection_in_adaptation_set) {
+  if (!adaptation_set.codec_.empty()) {
+    std::string adaptation_set_codec =
+        base::SplitString(adaptation_set.codec_, ".", base::TRIM_WHITESPACE,
+                          base::SPLIT_WANT_NONEMPTY)[0];
+
+    if (media_info.has_video_info() && media_info.video_info().has_codec()) {
+      if (adaptation_set_codec.compare(base::SplitString(
+              media_info.video_info().codec(), ".", base::TRIM_WHITESPACE,
+              base::SPLIT_WANT_NONEMPTY)[0]) != 0)
+        return false;
+
+    } else if (media_info.has_audio_info() &&
+               media_info.audio_info().has_codec()) {
+      if (adaptation_set_codec.compare(base::SplitString(
+              media_info.audio_info().codec(), ".", base::TRIM_WHITESPACE,
+              base::SPLIT_WANT_NONEMPTY)[0]) != 0)
+        return false;
+
+    } else if (media_info.has_text_info() &&
+               media_info.text_info().has_codec()) {
+      if (adaptation_set_codec.compare(base::SplitString(
+              media_info.text_info().codec(), ".", base::TRIM_WHITESPACE,
+              base::SPLIT_WANT_NONEMPTY)[0]) != 0)
+        return false;
+    }
+  }
```

Comment:
Is SplitString needed? 

with the change suggested above in line 412 of adaptation_set.cc, we can just do:

if (adaptation_set.codec() != GetCodecs(media_info))
  return false;

Reply:
GetCodecs returns the entire codec string (avc1.4d400d / avc1.4d401f) if container type is not webm. Should I change the GetCodecs to return media_info.video_info().codec().substr(0, 4); for all container types or keep the SplitString in Match?

Reply:
Ah, good catch, we shouldn't be using the entire codec string as two different representations in the same adaptation set could have two different codec strings.

Please replace GetCodecs with GetBaseCodec instead.

Reply:
TrickPlayWithMatchingAdaptationSet unit test is failing as the codec in default_adaptation_set_ is no call to UpdateFromMediaInfo.  
So, instead of setting codec from AdaptationSet::UpdateFromMediaInfo do you think it would make sense to set it at Period::SetNewAdaptationSetAttributes. If not, how do you change the unit test? I tried calling the UpdateFromMediaInfo directly but that did not work.

Reply:
Sure. You may add a AdaptationSet::set_codec(const std::string& codec) function and call it in Period::SetNewAdaptationSetAttributes. 

(This tells me that we should really refactor the code later to clean up the code and remove ProtectedContentMap)